cli/pcluster/cfnconfig.py

```diff
@@ -545,6 +548,26 @@ def __init_efa_parameters(self):
         except configparser.NoOptionError:
             pass
 
+    def __init_hyperthreading_parameter(self):
+        try:
+            __temp__ = self.__config.get(self.__cluster_section, "disable_hyperthreading")
+            if __temp__ not in  {"true", "false"}:
+                self.__fail("valid values for disable_hyperthreading = true/false")
+            # TODO: validate if instance type supports CPU Options
+            if __temp__ == "true":
+                master_instance_type = self.parameters.get("MasterInstanceType", "t2.micro")
+                compute_instance_type = self.parameters.get("ComputeInstanceType", "t2.micro")
+                master_vcpus = get_instance_vcpus(self.region, master_instance_type)
+                compute_vcpus = get_instance_vcpus(self.region, compute_instance_type)
+                if master_vcpus < 1 or compute_vcpus < 1:
+                    return # TODO: throw exception?
+                master_core_count = int(master_vcpus) // 2
+                compute_core_count = int(compute_vcpus) // 2
+                self.parameters["MasterCoreCount"] = str(master_core_count)
```

Comment:
do you need to typecast here since the cloudformation has type Number?

Reply:
Yes the template still seems to except a string to be passed